examples/weather-demo/src/weather/openweathercontroller.rs

```diff
@@ -0,0 +1,417 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: MIT
+
+#![cfg(not(target_arch = "wasm32"))]
+
+use chrono::DateTime;
+use openweather_sdk::responses::{GeocodingResponse, OneCallResponse};
+use openweather_sdk::{Language, OpenWeather, Units};
+use serde::{Deserialize, Serialize};
+use std::fs::File;
+use std::io;
+use std::io::{BufReader, BufWriter, Write};
+use std::ops::Deref;
+use std::path::PathBuf;
+use std::sync::Arc;
+use std::vec;
+use tokio::sync::Mutex;
+
+use crate::weather::utils::*;
+use crate::weather::weathercontroller::{
+    CityData, CityWeatherData, DayWeatherData, ForecastWeatherData, GeoLocationData,
+    PrecipitationData, TemperatureData, WeatherCondition, WeatherController, WeatherData,
+};
+
+#[cfg(target_os = "android")]
+use std::env;
+
+const CITIES_STORED_FILE_NAME: &str = "cities_data.json";
+const ORGANIZATION_QUALIFIER: &str = "com"; // have to match android app name in cargo.toml
+const ORGANIZATION_NAME: &str = "felgo.demos"; // have to match android app name in cargo.toml
```

Comment:
Maybe this needs changing/adapting? Just a drive-by thought.

Reply:
I guess it makes sense, but let me double-check this and get back to you :slightly_smiling_face:

Reply:
Sorry for the delay. I was offline last week.
I changed this. Please let me know if the new version is okay with you.